h2o-extensions/xgboost/src/main/java/ml/dmlc/xgboost4j/java/XGBoostScoreTask.java

```diff
@@ -166,7 +180,7 @@ public void map(Chunk[] cs, NewChunk[] ncs) {
                 double[] dpreds = new double[preds.length];
                 for (int j = 0; j < dpreds.length; ++j)
                     dpreds[j] = preds[j][0];
-                for (int i = 0; i < cs[0]._len; ++i) {
+                for (int i = 0; i < (weightColumnIndex != -1 ? weights.length : cs[0].len()); ++i) {
```

Comment:
This looks suspicious, all the chunks should have the same length. Why do you need to take the `weights` as the reference?